### ISSUE TITLE
Not support database table prefix name.

### ISSUE THREAD

**Issue by hgati** [yireo/Yireo_DeleteAnyOrder2#5]
This extension is unable to retrieve the Magento database prefix. Therefore, users who utilize prefix table names may encounter the following error:

```
3 exception(s):
Exception #0 (Magento\Framework\DB\Adapter\TableNotFoundException): SQLSTATE[42S02]: Base table or view not found: 1146 Table 'magento.sales_order' doesn't exist, query was: SELECT `entity_id` FROM sales_order
Exception #1 (Zend_Db_Statement_Exception): SQLSTATE[42S02]: Base table or view not found: 1146 Table 'magento.sales_order' doesn't exist, query was: SELECT `entity_id` FROM sales_order
Exception #2 (PDOException): SQLSTATE[42S02]: Base table or view not found: 1146 Table 'magento.sales_order' doesn't exist

Exception #0 (Magento\Framework\DB\Adapter\TableNotFoundException): SQLSTATE[42S02]: Base table or view not found: 1146 Table 'magento.sales_order' doesn't exist, query was: SELECT `entity_id` FROM sales_order
<pre>#1 Magento\Framework\DB\Adapter\Pdo\Mysql->query() called at [vendor/yireo/magento2-deleteanyorder2/Fixer/DataProvider/OrderId.php:39]
#2 Yireo\DeleteAnyOrder2\Fixer\DataProvider\OrderId->getData() called at [vendor/yireo/magento2-deleteanyorder2/Fixer/Fixer.php:77]
#3 Yireo\DeleteAnyOrder2\Fixer\Fixer->getCurrentOrderIds() called at [vendor/yireo/magento2-deleteanyorder2/Fixer/Fixer.php:127]
#4 Yireo\DeleteAnyOrder2\Fixer\Fixer->getOrphansCountPerTable() called at [vendor/yireo/magento2-deleteanyorder2/ViewModel/Fixer.php:47]
#5 Yireo\DeleteAnyOrder2\ViewModel\Fixer->getOrphansCountPerTable() called at [vendor/yireo/magento2-deleteanyorder2/view/adminhtml/templates/overview.phtml:31]
```

**Comment by jissereitsma** [comment]
Thanks for opening an issue. I noticed that you closed the issue right away. Could you please comment on why the issue was opened in the first place and why you closed it again? If there is anything in this extension that could be improved, it would be nice to know.

**Comment by hgati** [comment]
> Thanks for opening an issue. I noticed that you closed the issue right away. Could you please comment on why the issue was opened in the first place and why you closed it again? If there is anything in this extension that could be improved, it would be nice to know.

My Magento database prefix starts with cb_, but the error message is missing the prefix.
I have closed it as I don't have the time to analyze this issue right now.

**Comment by jissereitsma** [comment]
Understood. But opening an issue does not require you to fix the issue. I didn't know about this issue yet. However, after simple testing, I'm able to confirm this. And provide a fix. A new release 1.0.7 should now support table prefixes.

**Comment by hgati** [comment]
Okay, I have reopened and will investigate further.

**Comment by hgati** [comment]
I've installed with composer require command. but the version is 1.0.5
![](https://i.ibb.co/yRMr87g/001.png)

**Comment by hgati** [comment]
![](https://i.ibb.co/WtKTpGg/002.png)

It seems that the package repository hasn't been updated yet on Packgist.

**Comment by hgati** [comment]
![](https://i.ibb.co/YLMyS7m/003.png)


I am still encountering an error.
```
3 exception(s):
Exception #0 (Magento\Framework\DB\Adapter\TableNotFoundException): SQLSTATE[42S02]: Base table or view not found: 1146 Table 'magento.sales_creditmemo' doesn't exist, query was: SELECT COUNT(`order_id`) FROM `sales_creditmemo`
Exception #1 (Zend_Db_Statement_Exception): SQLSTATE[42S02]: Base table or view not found: 1146 Table 'magento.sales_creditmemo' doesn't exist, query was: SELECT COUNT(`order_id`) FROM `sales_creditmemo`
Exception #2 (PDOException): SQLSTATE[42S02]: Base table or view not found: 1146 Table 'magento.sales_creditmemo' doesn't exist

Exception #0 (Magento\Framework\DB\Adapter\TableNotFoundException): SQLSTATE[42S02]: Base table or view not found: 1146 Table 'magento.sales_creditmemo' doesn't exist, query was: SELECT COUNT(`order_id`) FROM `sales_creditmemo`
<pre>#1 Magento\Framework\DB\Adapter\Pdo\Mysql->query() called at [vendor/magento/zend-db/library/Zend/Db/Adapter/Abstract.php:828]
#2 Zend_Db_Adapter_Abstract->fetchOne() called at [vendor/yireo/magento2-deleteanyorder2/Fixer/Fixer.php:152]
#3 Yireo\DeleteAnyOrder2\Fixer\Fixer->getTotalsPerTable() called at [vendor/yireo/magento2-deleteanyorder2/ViewModel/Fixer.php:57]
#4 Yireo\DeleteAnyOrder2\ViewModel\Fixer->getTotalsPerTable() called at [vendor/yireo/magento2-deleteanyorder2/view/adminhtml/templates/overview.phtml:30]
#5 include() called at [vendor/magento/framework/View/TemplateEngine/Php.php:71]
#6 Magento\Framework\View\TemplateEngine\Php->render() called at [vendor/magento/framework/Interception/Interceptor.php:58]
#7 Magento\Framework\View\TemplateEngine\Php\Interceptor->___callParent() called at [vendor/magento/framework/Interception/Interceptor.php:138]
#8 Magento\Framework\View\TemplateEngine\Php\Interceptor->Magento\Framework\Interception\{closure}() called at [vendor/magento/framework/Interception/Interceptor.php:153]
```

**Comment by hgati** [comment]
Debug based on above error message.
![](https://i.ibb.co/N7SzczN/005.png)

The error occurs because the prefix is missing in the error section.
![](https://i.ibb.co/FY2zx67/006.png)

**Comment by hgati** [comment]
Obtaining the table prefix overall is challenging. The module's design is intertwined with various custom classes, making it difficult to simply retrieve the table prefix. I attempted to retrieve the table prefix easily but eventually gave up due to this entanglement.

![](https://i.ibb.co/N1J0523/007.png)

I successfully retrieved the table prefix, but a different error is occurring.
```
TypeError: Yireo\DeleteAnyOrder2\Fixer\TableFactory::create(): Return value must be of type Yireo\DeleteAnyOrder2\Fixer\AbstractTable, string returned in /var/www/magento/vendor/yireo/magento2-deleteanyorder2/Fixer/TableFactory.php:40
Stack trace:
#0 /var/www/magento/vendor/yireo/magento2-deleteanyorder2/Fixer/Fixer.php(200): Yireo\DeleteAnyOrder2\Fixer\TableFactory->create()
#1 /var/www/magento/vendor/yireo/magento2-deleteanyorder2/ViewModel/Fixer.php(37): Yireo\DeleteAnyOrder2\Fixer\Fixer->getTables()
#2 /var/www/magento/vendor/yireo/magento2-deleteanyorder2/view/adminhtml/templates/overview.phtml(9): Yireo\DeleteAnyOrder2\ViewModel\Fixer->getTables()
#3 /var/www/magento/vendor/magento/framework/View/TemplateEngine/Php.php(71): include('...')
```

**Comment by hgati** [comment]
This issue arises due to the lack of support for table prefixes.
So I've uninstalled again.
```
bin/magento module:uninstall Yireo_DeleteAnyOrder2
```

**Comment by jissereitsma** [comment]
I'm not sure if I follow you. You mentioned "I have reopened and will investigate further" - but I hope you do understand that again, opening an issue does **not** mean fixing it. I already had a fix as mentioned, version 1.0.7, but apparently it was not pushed to Packagist yet. So, as I see it, you started to dig into code, found it overwhelming, while actually I already went through the trouble of fixing it for you and everyone else. 

Removing the module is fine. I just hope that in the future, you can help open source projects by **only** reporting an issue, and then possibly investigating it, and perhaps even by adding a PR. But adding an issue, doing an investigation on code you don't understand, and then uninstalling it and giving up, is probably a waste of both your time and mine. Then it would have been better if you would have uninstalled the module if it was not working for you.

Anyway, as I see it, the problem was fixed by me, just not fast enough to serve you.

**Comment by hgati** [comment]
> 

I am not a developer but an operator, so I don't have the time to dedicate to module analysis. With a lot of daily tasks involved in business operations, I'm seeking modules that can be installed and work without errors right away. That's why I closed the issue. I attempted a quick fix by modifying only the table prefix, but it resulted in a different error, leading me to close it. I hope you understand.
